### DISCUSSION TITLE
useWatch on a field array of controlled component leads to unexpected behaviour

### DISCUSSION THREAD

**Discussion by nhanpdnguyen** [react-hook-form/react-hook-form#3336] [Q&A] [answered]
**Description**

Hi everyone, I'm implementing a feature using field array with some controlled custom input (using `render` attribute on `Controller`)

Then I ran into this issue, the `watchedFieldArray`'s newly-added item gets undefined. I noticed that this issue happens when inserting an item at an in-between index.

I am wondering if I've misused something, please take a look, thank you very much.

**Here's the CodeSandbox link**: https://codesandbox.io/embed/unruffled-thompson-yx492?file=/src/App.js:1628-2152.

**Step to reproduce:**

- Go to CodeSandbox link
- Click `Insert at index 1`
- Watch the Console logs and see the index-1 item is `undefined`
- Click `Append`
- Watch the Console logs and see the last item is not `undefined`

**Desired output:** the index-1 item is not `undefined`

**Comment by nhanpdnguyen** [comment, marked as answer]
Asked on [Spectrum](https://spectrum.chat/react-hook-form/help/usewatch-on-a-field-array-leads-to-unexpected-behaviour~6db3cb13-57d5-4e07-8c56-312cf38300ae) and got a reply to track the issue [here](https://github.com/react-hook-form/react-hook-form/issues/3353).

**Reply by bluebill1049** [reply]
yep, working on it.
